client/src/components/Common/BookingForm/BookingForm.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react'
+import Calender from '../../../Images/calendar_icon.png'
+import Clock from '../../../Images/clock_icon.png'
+import RoomIcon from '../../../Images/room_icon.png'
+import './bookingForm.css'
+import axios from 'axios'
+import { Link } from 'react-router-dom'
+import Popup from './Popup'
+class BookingForm extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      userId : 3,
+      date: '2020-05-23T11:00:00.000Z',
+      roomId:  5,
+      room_name : 'Moscow' ,
+      toggelReminder: false ,
+      showPopup : false
+    }
+  }
+
+  handelChange = (e) => {
+    const name = e.target.name
+    const value =e.target.value
+    this.setState({[name]: value})
+  }
+  handelChecked = ()=> {
+      this.setState({toggelReminder : !this.state.toggelReminder })
+  }
+
+   handelSubmit =(e) => {
+      e.preventDefault();
+      const formData = this.state
+      axios.post( '/api/booking', formData)
+      .then(response => {if(response.status===200){
+        console.log( 'res :',response)
+       return this.setState({showPopup : !this.state.showPopup})}
+      // else  alert('A small error happend in the server in this booking operation!! try again !!')
+      })
```

Comment:
several things to this:

1. code formatting! This is difficult to read. The `if` on line 35 should be on a new line, the last `}` on line 37 should also be on a separate line; plus there is inconsistent indentation. Also, pls remove the commented code. Here is how it should look (there should be an autoformat feature in your IDE you can use if you want)
```js
axios.post( '/api/booking', formData)
  .then(response => {
    if(response.status===200) {
      console.log( 'res :',response)
      return this.setState({showPopup : !this.state.showPopup})
    }
  })
```
2. On line 37, the `setState()` method does not return any value, thus you can remove the `return` statement. Let me know on Discord if it's not clear why, the promises and the asynchronous nature of React and `setState` can be difficult to understand at first..
3. The `if(response.status === 200) {` check can actually be removed entirely. If you check Axios' [documentation](https://github.com/axios/axios) (look for `validateStatus`), it does this check automatically by default and will throw an error if the status is not in the 200 to 300 range. What that means is that in the `then` function of the promise you are guaranteed that the status is within 200, thus no need to check. For any other status, the promise will automatically throw an error and it will end up in the `catch` function
4. Lastly, I suggest you move the Axios part of this code to a new module, say `src/api/index.js` and have a function in there for each API you use. Why? Imagine you call this same API from 3 other components. Suddenly you decide to change the URL, or the parameters. You will have to go to every place where this is called and do the change. This is code duplication and error prone. So, there should be a module with all the APIs defined as functions:
```js
function bookEvent(formData) {
  return axios.post('/api/booking', formData)
}
```
and then call it from here as 
```js
api.bookEvent(formData)
  .then()
  .catch()
```